pkg/app/pipedv1/livestatereporter/livestatereporter.go

```diff
@@ -45,127 +51,225 @@ type Reporter interface {
-		})
+		snapshotFlushInterval: 1 * time.Minute,
+		appLister:             appLister,
+		apiClient:             apiClient,
+		gitClient:             gitClient,
+		repoMap:               make(map[string]git.Repo),
+		pluginRegistry:        pluginRegistry,
+		pipedConfig:           pipedConfig,
+		secretDecrypter:       secretDecrypter,
+		workingDir:            workingDir,
+		logger:                rlogger,
 	}
-	return r
+
+	return r, nil
 }
 
 func (r *reporter) Run(ctx context.Context) error {
-	group, ctx := errgroup.WithContext(ctx)
+	r.logger.Info("start running app live state reporter", zap.Duration("snapshot-flush-interval", r.snapshotFlushInterval))
 
-	r.logger.Info(fmt.Sprintf("starting %d live state reporters of plugins", len(r.reporters)))
+	snapshotTicker := time.NewTicker(r.snapshotFlushInterval)
+	defer snapshotTicker.Stop()
 
-	for _, reporter := range r.reporters {
-		// Avoid starting all reporters at the same time to reduce the API call burst.
-		time.Sleep(10 * time.Second)
+	for {
+		select {
+		case <-snapshotTicker.C:
+			r.flushSnapshots(ctx)
 
-		group.Go(func() error {
-			reporter.logger.Info("starting app live state reporter for plugin")
-			return reporter.Run(ctx)
-		})
+		case <-ctx.Done():
+			r.logger.Info("app live state reporter has been stopped")
+			return nil
+		}
 	}
+}
 
-	r.logger.Info(fmt.Sprintf("all live state reporters of %d plugins have been started", len(r.reporters)))
+func (r *reporter) flushSnapshots(ctx context.Context) {
+	r.logger.Info("start flushing snapshots")
 
-	if err := group.Wait(); err != nil {
-		r.logger.Error("failed while running", zap.Error(err))
-		return err
+	r.logger.Info("updating git repositories")
+	for id, cfgRepo := range r.pipedConfig.GetRepositoryMap() {
+		if _, ok := r.repoMap[id]; ok {
+			r.logger.Info("the repo is cloned, so pulling repository", zap.String("repo-id", id))
+			err := r.repoMap[id].Pull(ctx, cfgRepo.Branch)
+			if err != nil {
+				r.logger.Error("failed to pull repository", zap.String("repo-id", id), zap.Error(err))
+			}
+			continue
+		}
+
+		r.logger.Info("the repo is not cloned, so cloning repository", zap.String("repo-id", id))
+		repo, err := r.gitClient.Clone(ctx, id, cfgRepo.Remote, cfgRepo.Branch, fmt.Sprintf("%s/%s", r.workingDir, id))
+		if err != nil {
+			r.logger.Error("failed to clone repository", zap.String("repo-id", id), zap.Error(err))
+			continue
+		}
+		r.repoMap[id] = repo
 	}
 
-	r.logger.Info(fmt.Sprintf("all live state reporters of %d plugins have been stopped", len(r.reporters)))
-	return nil
-}
+	apps := r.appLister.List()
 
-type pluginReporter struct {
-	pluginName            string
-	snapshotFlushInterval time.Duration
-	appLister             applicationLister
-	apiClient             apiClient
-	pluginClient          pluginapi.PluginClient
-	logger                *zap.Logger
-}
+	// TODO: Set the limit based on the piped config
+	limit := runtime.GOMAXPROCS(0) / 2
```

Comment:
When `runtime.GOMAXPROCS(0)` returns `1`, `limit` becomes `0`.
This causes the division-by-zero error.
Please set the limit `1` if it is `0`.

Reply:
Thanks, fixed on [5ea0571](https://github.com/pipe-cd/pipecd/pull/5457/commits/5ea057119e7a07f137e7308f83c9f430a200e555)